charts/merlin/templates/deployment.yaml

```diff
@@ -79,17 +79,26 @@ spec:
             value: "{{ .Values.merlin.loggerDestinationURL }}"
           - name: PORT
             value: "{{ .Values.merlin.service.internalPort }}"
+          {{- if .Values.postgresql.enabled }}
```

Comment:
instead of having conditions here, i prefer to use _helpers.tpl

in _helpers.tpl, I will have this:

```
{{- define "database.host" -}}
{{- if .Values.postgresql.enabled -}}
{{- printf "%s-postgresql.%s.svc.cluster.local" .Release.Name .Release.Namespace -}}
{{- else -}}
{{- .Values.merlin.database.host -}}
{{- end -}}
{{- end -}}

and 3 more for db name, user, and password
```

then in deployment.yaml, there will be no logic, but simply:

```
- name: DATABASE_HOST
   value: {{ template "database.host" . }}
```

Reply:
At first, I plan to do this however after implementing it, the helper's class become bloated.
Meanwhile, I have seen some logic in the templates like the sentry enable, etc. It is not hard to read.

WDYT?

I don't mind to revert it back to helpers template though

Reply:
Sentry's case is different from database config. For sentry, we simply check if we are going to use it or not. But for database, we need to decide which value to use for a given postgresql provider.

_helpers.tpl supposed to be "bloated" because it's abstract the complexity of other template files.

Reply:
Ah understood, I saw similar pattern in postgresql as well mas.
Thank you for the review already refactor it